File: solutions/javascript/isogram/1/isogram.js
/**
 * Checks if a given string is an isogram.
 * An isogram is a word or phrase without a repeating letter, 
 * ignoring spaces and hyphens.
 *
 * @param {string} string - The input string to be checked.
 * @returns {boolean} - Returns true if the string is an isogram, false otherwise.
 */
export const isIsogram = (string) => {
  // Remove spaces and hyphens from the string
  const stringNoSpaceOrHyphen = string.replace(/[\s-]/g, '');

  // Convert string to lowercase and split into individual characters
  const characters = stringNoSpaceOrHyphen.toLowerCase().split('');

  // Check if all characters are unique
  const uniqueCharacters = new Set(characters);

  // If the size of the set matches the length of the string, it's an isogram
  return uniqueCharacters.size === characters.length;
};

File: solutions/python/isogram/2/isogram.py
"""
This module provides a function to check if a given string is an isogram.

An isogram is a word or phrase without a repeating letter, however spaces and hyphens are allowed to appear multiple times.
"""

import re


def is_isogram(string):
    """
    Check if the input string is an isogram.

    An isogram is a word or phrase without a repeating letter. Spaces and hyphens are ignored.

    Parameters:
    string (str): The string to check.

    Returns:
    bool: True if the string is an isogram, False otherwise.
    """
    string = re.sub(r"[ -]", "", string.lower())
    return len(string) == len(set(string))
